src/classes/frDonorTest.cls

```diff
@@ -113,6 +112,50 @@ public class frDonorTest {
         System.assertEquals(String.valueOf(requestBody.get('institutionName')), newContact.LastName, 'We expected a fallback so that the contact would still get created');
     }
     
+    //Sync to an existing contact that once the mappings are applied, LastName is empty so Salesforce will throw an exception
+    //confirm our fallback works as expected to prevent that exception
+    static testMethod void syncEntity_contactLastNameFallback_existing() {
+		Contact existingSupporter = getTestContact();
+        insert existingSupporter;
+        
+        RestRequest req = new RestRequest();
+        RestResponse res = new RestResponse();
+        
+        req.requestURI = 'https://XXXX.salesforce.com/services/apexrest/funraise/v1/donor';
+        req.httpMethod = 'POST';
+        Map<String, Object> requestBody = getTestRequest();
+        requestBody.put('id', existingSupporter.fr_Id__c);
+        requestBody.put('lastName', null);
+        requestBody.put('institutionName', 'Test Inst Name');
+        req.requestBody = Blob.valueOf(Json.serialize(requestBody));
+        RestContext.request = req;
+        RestContext.response = res;
+        
+        createMapping('firstName', 'FirstName');
+        createMapping('lastName', 'LastName');
```

Comment:
This test won't catch the issue in the original bug, but I think it captures the original spirit of the fallback, which is to catch the part where even after we apply the mappings (and there were mappings for all of the required Salesforce fields), the values from Funraise meant that we would get an exception from Salesforce.  This test confirms that under normal circumstances, a request from Funraise that is missing required information will not throw an exception